Fixed three bugs in hclapFilter

sts1skj opened the pull request:
1.  hclapFilter was not handling welded vertices properly.  This resulted
in smoothing brushes unsmoothing such vertices instead of smoothing them.

2.  hclapFilter did not correctly implement the HC Laplacian filter.
It was using the previous iteration's values where it was supposed to use
the original values (where multiplied by hcAlpha).

3.  hclapFilter was using updated values for some (but not all)
points instead of the previous iteration's values (where multiplied
by 1-hcAlpha).

sts1skj committed:
Fixed three bugs in hclapFilter

1.  hclapFilter was not handling welded vertices properly.  This resulted
in smoothing brushes unsmoothing such vertices instead of smoothing them.

2.  hclapFilter did not correctly implement the HC Laplacian filter.
It was using the previous iteration's values where it was supposed to use
the original values (where multiplied by hcAlpha).

3.  hclapFilter was using updated values for some (but not all)
points instead of the previous iteration's values (where multiplied
by 1-hcAlpha).

sts1skj commented:
@ousnius: I could also templatize lapFilter and hclapFilter, if you like, so that there's one copy of each instead of three.

ousnius commented:
Smoothing seems to work way better now, thanks! If the templating doesn't get too complicated/confusing, go ahead. Otherwise I'd prefer more readable code to templates.

PS: There's some warnings in MSVC++ about unused variables (int a) and multiple declarations for int i.

ousnius commented on src/components/TweakBrush.cpp:
```diff
@@ -511,28 +511,17 @@ TB_SmoothMask::~TB_SmoothMask() {
-	Vector3 d;
 	int adjPoints[1000];
 	int c = 0;
 	int a;
```
Unused variable, there's multiple like these.

sts1skj committed:
Removed unused variables from TweakBrush.cpp.

The last commit left some unused variables around and some places
where one variable hid a variable in a larger scope.  I've fixed those
to silence the MSVC compiler warnings.

sts1skj commented:
Hopefully I've fixed those warnings from MSVC.

sts1skj commented:
I'm giving up on templatizing lapFilter and hclapFilter.  I don't see how to do it nicely.
